src/lavinmqctl.cr

```diff
@@ -700,6 +704,56 @@ class LavinMQCtl
     data_dir = ARGV.shift? || abort "definitions <datadir>"
     DefinitionsGenerator.new(data_dir).generate(STDOUT)
   end
+
+  private def cleanup_datadir
+    data_dir = ARGV.shift? || abort "cleanup_datadir <datadir>"
+    vhost = ARGV.shift?
+
+    vhosts = [vhost]
+    unless vhost # no vhost specified, get all vhosts
+      resp = http.get "/api/vhosts", @headers
+      handle_response(resp, 200)
+      if vhost_list = JSON.parse(resp.body).as_a?
+        vhosts = vhost_list.map do |u|
+          next unless v = u.as_h?
+          v["name"].to_s
+        end
+      end
+    end
+
+    vhosts.compact.each do |current_vhost|
+      resp = http.get "/api/queues/#{URI.encode_www_form(current_vhost)}", @headers
+      handle_response(resp, 200)
+      queue_list = [] of String
+      if queues = JSON.parse(resp.body).as_a?
+        queue_list = queues.map do |q|
+          next unless v = q.as_h?
+          Digest::SHA1.hexdigest(v["name"].to_s)
+        end
+      end
```

Comment:
You'll need to make loop here to make sure you get all queues since the result is paginated. It's probably good to use `LavinMQ::HTTP::Controller::MAX_PAGE_SIZE` as page size to reduce the number of calls.

Reply:
Ah. That seems to be true for pretty much all GETs in lavinmqctl 😬 
I'll take a look at that!

Reply:
This should fix that for all GETs https://github.com/cloudamqp/lavinmq/pull/904